Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/PerfLib.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Globalization;
+
+    internal class PerfLib : IQuickPulsePerfLib
+    {
+        private static PerfLib library = null;
+
+        private PerformanceMonitor performanceMonitor;
+
+        private PerfLib()
+        {
+        }
+
+        public static PerfLib GetPerfLib()
+        {
+            library = library ?? new PerfLib();
```

Comment:
We could. The reason why we're creating an instance of this here is because we don't want the higher levels to deal with this low-level detail, while at the same time we want a singleton. Plus there's a nice lazy initialization benefit.